### PR TITLE
:bug: Add missing `cancel` event to `<input type=file>`

### LINKED ISSUE #22157: Document HTMLInputElement's (type=file) cancel event
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement

### What specific section or headline is this issue about?

Events

### What information was incorrect, unhelpful, or incomplete?

The `cancel` event is missing.

### What did you expect to see?

A new page for the cancel event with an example.

### Do you have any supporting links, references, or citations?

https://bugzilla.mozilla.org/show_bug.cgi?id=1719703
https://html.spec.whatwg.org/#show-the-picker,-if-applicable
https://html.spec.whatwg.org/#event-cancel

### Do you have anything more you want to share?

_No response_

### PR DESCRIPTION
**Pull request by FurryR**
### Description

Add `cancel` event to `<input type=file>`.

### Motivation

👎 People are using hacks (`window.addEventListener('focus')`) to detect cancellations of file dialogs.

[An example from GoogleChromeLabs/browser-fs-access](https://github.com/GoogleChromeLabs/browser-fs-access/blob/5d8a551af106121d010c1c50ca4a15d3cb0b189d/src/legacy/file-open.mjs#L32-L45)

However, in [https://html.spec.whatwg.org/multipage/indices.html#event-cancel](https://html.spec.whatwg.org/multipage/indices.html#event-cancel):

> Fired at [dialog](https://html.spec.whatwg.org/multipage/interactive-elements.html#the-dialog-element) elements when they are canceled by the user (e.g., by pressing the Escape key), or at [input](https://html.spec.whatwg.org/multipage/input.html#the-input-element) elements in the [File](https://html.spec.whatwg.org/multipage/input.html#file-upload-state-(type=file)) state when the user does not change their selection

It helps a lot.

```javascript
function select_file() {
  return new Promise((resolve, reject) => {
    const elem = document.createElement('input')
    elem.type = 'file'
    elem.addEventListener('cancel', () => {
      reject(new Error('Cancelled')) // hooray!
    })
    elem.addEventListener('change', () => {
      if (elem.files.length == 1) resolve(elem.files.item(0))
      else reject(new Error('Cancelled'))
    })
    elem.click()
  })
}
```

### Additional details

Reference: [https://html.spec.whatwg.org/multipage/indices.html#event-cancel](https://html.spec.whatwg.org/multipage/indices.html#event-cancel)

### Related issues and pull requests

🔨 Fixes #22157.

### CODE REVIEW

**Comment by estelle** [comment]
Before adding a link to the cancel event in the technical summary, the page needs to be created. See my [comment in the issue](https://github.com/mdn/content/issues/22157#issuecomment-1764985663)

**Comment by FurryR** [comment]
> Before adding a link to the cancel event in the technical summary, the page needs to be created. See my [comment in the issue](https://github.com/mdn/content/issues/22157#issuecomment-1764985663)

Sorry for replying so late, I will create the page later.

**Review by estelle — COMMENTED** [review]
I made some suggested edits. I think we need to be clearer about the multiple ways the cancel event may be thrown.

**Comment by FurryR** [comment]
@estelle suggested changes are all merged. Don't mind the commit messages, I forgot that `#1` means #1.

File: files/en-us/web/api/htmlelement/cancel_event/index.md [hunk 1]
```diff
@@ -80,7 +80,46 @@ closeButton.addEventListener("click", () => {
...
+elem.addEventListener("change", () => {
+  if (elem.files.length == 1) {
+    result.textContent = "File Selected.";
+  }
+});
+```
+
+#### Result
+
+{{ EmbedLiveSample('input_cancel', '100%', '100px') }}
```
Comment (estelle) [inline files/en-us/web/api/htmlelement/cancel_event/index.md:122]:
```suggestion
{{ EmbedLiveSample('input_cancel', '100%', '100px') }}

Open the file selector, then close the selection dialog with the escape key or the cancel button. Both of these will cause the cancel event to be fired. Also, try selecting a local file on your machine; then reopen the file selection window and reselect the same file. This too causes the cancel event to be fired.
```

**Review by estelle — COMMENTED** [review]
I find the behavior of reselecting the same files unexpected, so want to make sure that it's clear. I made two suggestions to emphasize that. I am not sure if my suggestions are overkill. 

also, when you're ready for a re-review, click the little icon next to my name so that you see a small brown dot instead of the icon. This informs me that the PR is ready for re-review.

**Comment by FurryR** [comment]
Merged

**Review by estelle — APPROVED** [review]
THank you!

**Comment by skyclouds2001** [comment]
I think instead of moving page from `HTMLDialogElement: cancel` to `HTMLElement: cancel`, it is better to create a page for `HTMLInputElement: cancel`, this is also the behavior of BCD

**Pull request merged**